include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -6595,18 +6592,18 @@ namespace internal
 
     const unsigned int temp_size = Eval::dofs_per_cell > Eval::n_q_points ?
                                    Eval::dofs_per_cell : Eval::n_q_points;
-    VectorizedArray<Number>  temp_data[temp_size < 100 ? 2*temp_size : 1];
+    VectorizedArray<Number>  temp_data[(temp_size > 0 && temp_size < 100) ? 2*temp_size : 1];
     VectorizedArray<Number> *temp1;
     VectorizedArray<Number> *temp2;
-    if (temp_size < 100)
+    if (temp_size > 0 && temp_size < 100)
       {
         temp1 = &temp_data[0];
         temp2 = temp1 + temp_size;
       }
     else
       {
         temp1 = scratch_data;
-        temp2 = scratch_data + (temp_size < numbers::invalid_unsigned_int ? temp_size :
+        temp2 = scratch_data + (temp_size > 0 ? temp_size :
```

Comment:
I'm going to note that conceptually, we treat `numbers::invalid_unsigned_int` as a value that lies *outside* the normal numbers, in the same way as we do when considering `NaN`. As a consequence, I'm not fond of comparing for `<` with such denormals. I think we should always use `!=` because that makes no assumption on where exactly this denormal is located within the realm of numbers.